central/role/resources/list.go

```diff
@@ -38,26 +38,18 @@ var (
 	DeploymentExtension = newResourceMetadata("DeploymentExtension", permissions.NamespaceScope)
 	Detection           = newResourceMetadata("Detection", permissions.GlobalScope)
 	Image               = newResourceMetadata("Image", permissions.NamespaceScope)
-	// SAC check is not performed directly on ImageComponent resource. It exists here for postgres sac generation to pass.
-	ImageComponent = newResourceMetadata("ImageComponent", permissions.NamespaceScope)
```

Comment:
I don't think we can remove those.
Some (`ImageComponent`) are currently defined within [roleResources.ts](https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/types/roleResources.ts) and exposed to clients in the UI within [ResourceDescription.ts](https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/AccessControl/PermissionSets/ResourceDescription.tsx).
So, technically, clients are capable of choosing these for their permission sets, simply removing them would break them and we require a migration for that.

The second thing, even though they are not defined within UI, it's still possible to use those via API, so we would break existing clients here as well.

For this, we should define them as deprecated resource (`newDeprecatedResourceMetadata`), using `resource.Image/resource.Node` as replacing resource.

Please also mention the deprecation announcement in the CHANGELOG, luckily we still can manage with the 3.71 release. Additionally, would be great if you could create a follow-up task in JIRA that will cover the creation of a migration for permission sets that contain the old resources.